.github/workflows/check-providers.yml

```diff
@@ -14,31 +14,41 @@ on:
 jobs:
   check-providers:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [20]
+        pnpm-version: [8]
     steps:
       - name: Clone @api3/chains
         uses: actions/checkout@v4
 
-      - name: Setup Node
+      - name: Setup Node ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
-          node-version: '18'
-          cache: 'yarn'
+          node-version: ${{ matrix.node-version }}
 
-      - name: Get yarn cache directory path
-        id: deps-cache-dir-path
-        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      - uses: pnpm/action-setup@v3
+        name: Install pnpm ${{ matrix.pnpm-version }}
+        with:
+          version: ${{ matrix.pnpm-version }}
+          run_install: false
+
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
```

Comment:
(Comment also applies to the other workflows) - I think you can do away with all of the cache and store steps by using:

```
      uses: actions/setup-node@v4
      with:
        node-version: ${{ matrix.node-version }}
        cache: 'pnpm'
```
*after* the `pnpm/action-setup@v3` step per the [pnpm Actions docs](https://pnpm.io/continuous-integration#github-actions). Indeed that's what I have in my [OIS pnpm fork](https://github.com/dcroote/ois/blob/ecb4294323a0bab92c3bcae0cd58a40c236b8526/.github/workflows/main.yml#L27-L34) and the Action run [uses caching](https://github.com/dcroote/ois/actions/runs/8310212528/job/22742457482#step:4:11) ("Found in cache @ /opt/hostedtoolcache/node/18.19.1/x64")

Reply:
Ah nice, thanks 🚀